src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -213,7 +213,10 @@ Account GetThroughCacheCheckExists()
             bool isZero = balanceChange.IsZero;
             if (isZero)
             {
-                if (releaseSpec.IsEip158Enabled)
+                // this also works like this in Geth (they don't follow the spec ¯\_(*~*)_/¯)
+                // however we don't do it because of a consensus issue with Geth, just to avoid
+                // hitting non-existing account when substractin Zero-value from the sender
+                if (releaseSpec.IsEip158Enabled && !isSubtracting)
```

Comment:
only zero value substraction (in practice they only happen on system txs in the sender account). this is something we are porting from geth, not related to optimism, geth does exactly this for every account on every transaction. it would solve issues in eip4788 too.